src/main/java/com/jersey/resources/CustomerResource.java

```diff
@@ -43,6 +46,7 @@ public CustomerResource(CustomerDao customerDao){
      */
     @GET
     @Path("secure/customers/{id}")
+    @PreAuthorize("hasPermission(#id,'CustomerResource', 'ROLE_USER,ROLE_ADMIN')")
```

Comment:
Also writing uppercase words seperated by underscore is not a convention that we have followed anywhere else in the code.